dataset_loaders/videos/movingMNIST.py

```diff
@@ -0,0 +1,255 @@
+
+            # Compute the delta speed components along the (x, y) axis
+            speed_x += delta_x
+            speed_y += delta_y
+
+            # Take a step along velocity.
+            x += speed_x
+            y += speed_y
+            # Bounce off edges.
+            for j in range(self.num_digits):
+                if x[j] <= 0:
+                    x[j] = 0
+                    speed_x[j] = -speed_x[j]
+                elif x[j] >= 1.0:
+                    x[j] = 1.0
+                    speed_x[j] = -speed_x[j]
+                if y[j] <= 0:
+                    y[j] = 0
+                    speed_y[j] = -speed_y[j]
+                elif y[j] >= 1.0:
+                    y[j] = 1.0
+                    speed_y[j] = -speed_y[j]
+            trajectory_x[frame_id] = x
+            trajectory_y[frame_id] = y
+
+        # Scale to the size of the canvas.
+        trajectory_x = (canvas_size[1] * trajectory_x).astype(np.int32)
+        trajectory_y = (canvas_size[0] * trajectory_y).astype(np.int32)
+        return trajectory_x, trajectory_y
+
+    def _get_sequence(self, verbose=False):
+        trajectory_x, trajectory_y = self._get_random_trajectory()
+
+        # minibatch data
+        if self.random_background:
+            out_sequence = self._rng.rand(self.seq_length+1,
+                                          self.frame_size[0],
+                                          self.frame_size[1], 1)
+        else:
+            out_sequence = np.zeros((self.seq_length+1,
+                                     self.frame_size[0],
+                                     self.frame_size[1], 1),
+                                    dtype=np.float32)
+
+        for digit_id in range(self.num_digits):
+
+            # get random digit from dataset
+            curr_data_idx = self._rng.randint(
+                0, self._MNIST_data.shape[0]-1)
+            digit_image = self._MNIST_data[curr_data_idx]
+
+            zoom_factor = 1
+            if self.digits_sizes != 28:
+                zoom_factor = int(self.digits_sizes[digit_id]/28)
+                digit_image = zoom(digit_image, zoom_factor)
+            digit_size = digit_image.shape[0]
+
+            # generate video
+            digit_image = np.expand_dims(digit_image, -1)
+            import ipdb; ipdb.set_trace()
+            for i in range(self.seq_length+1):
+                top = trajectory_y[i, digit_id]
+                left = trajectory_x[i, digit_id]
+                bottom = top + digit_size
+                right = left + digit_size
+                out_sequence[i, top:bottom, left:right, :] = np.maximum(
+                    out_sequence[i, top:bottom, left:right, :], digit_image)
+        return out_sequence
+
+    def _reset(self, *args, **kwargs):
+        """ This function is used to reset the rng and data index from
+        where the digits are sampled to create the video sequence
+        """
+        if self.which_set != 'train':
```

Comment:
what is the reason for this condition?

Reply:
Ah ok, I got it: your intention was to suppress reset() to prevent the replacement of `self._rng` with the initial one, i.e., not to generate the same sequences again. 

What I'd suggest is rather to allow the user to request an infinite generator regardless of the set by setting `seq_per_subset` to `np.inf`. A few modifications are required for this to work:

1. Remove this line.
2. Edit https://github.com/fvisin/dataset_loaders/blob/master/dataset_loaders/parallel_loader.py#L445 as follows
```python
if self.seq_per_subset and self.seq_per_subset is not np.inf:
```
3. Edit https://github.com/fvisin/dataset_loaders/blob/master/dataset_loaders/parallel_loader.py#L41 --> "int or np.inf"
4. Edit https://github.com/fvisin/dataset_loaders/blob/master/dataset_loaders/parallel_loader.py#L467-L468 as follows:
```python
self.nsamples = len(names_sequences) if self.seq_per_subset is not np.inf else np.inf
self.nbatches = len(names_batches) if self.seq_per_subset is not np.inf else np.inf
```
5. Add at the beginning of `parallel_loaders.reset()`:
```python
if self.seq_per_subset and self.seq_per_subset is np.inf:
    import warnings
    warnings.warn("You cannot reset an infinite dataset (seq_per_subset is inf)!")
    return
```
6. Edit https://github.com/fvisin/dataset_loaders/blob/master/dataset_loaders/parallel_loader.py#L870 as follows:
```python
if self.seq_per_subset and self.seq_per_subset is np.inf:
    batch_to_load = 'inf'
else:
    batch_to_load = self.names_queue.get(False)
```
7. Edit https://github.com/fvisin/dataset_loaders/blob/master/dataset_loaders/parallel_loader.py#L540 as follows:
```python
if self.seq_per_subset and self.seq_per_subset is np.inf:
    name_batch = 'inf'
else:
    name_batch = self.names_batches.next()
```

Reply:
Please, **make sure these modifications work as intended!!!**
Check that 

- at least one of the `test()` methods of one of the other dataset work after you modify parallel_loaders
- check that self._reset and self._fill_names_batches are only called once at the initialization of the dataset when self.seq_per_subset is inf.

Reply:
Finally, you should also make sure that `dataset_instance.reset()` calls `self._reset()`, right?
It should suffice to define reset(), which should call self._reset() and then the super.